backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendHostCpuPinningsResource.java

```diff
@@ -0,0 +1,30 @@
+package org.ovirt.engine.api.restapi.resource;
+
+import org.ovirt.engine.api.model.HostCpuPinning;
+import org.ovirt.engine.api.model.HostCpuPinnings;
+import org.ovirt.engine.api.resource.HostCpuPinningsResource;
+import org.ovirt.engine.core.common.businessentities.VDS;
+import org.ovirt.engine.core.common.queries.IdQueryParameters;
+import org.ovirt.engine.core.common.queries.QueryType;
+import org.ovirt.engine.core.compat.Guid;
+
+public class BackendHostCpuPinningsResource extends AbstractBackendCollectionResource<HostCpuPinning, Object> implements HostCpuPinningsResource {
+
+    private Guid hostId;
+
+    public BackendHostCpuPinningsResource(Guid hostId) {
+        super(HostCpuPinning.class, Object.class);
+        this.hostId = hostId;
+    }
+
+    @Override
+    public HostCpuPinnings list() {
+        HostCpuPinnings pinnings = getMapper(VDS.class, HostCpuPinnings.class).map(getVdsByVdsId(), null);
+        pinnings.getHostCpuPinnings().stream().forEach(pinning -> addLinks(pinning));
```

Comment:
for your consideration: `pinnings.getHostCpuPinnings().forEach(this::addLinks);`